docs/content/querying/groupbyquery.md

```diff
@@ -207,60 +236,53 @@ indexing mechanism, and runs the outer query on these materialized results. "v2"
+|`maxOnDiskStorage`|`maxOnDiskStorage`|Maximum amount of disk space to use, per-query, for spilling result sets to disk when either the merging buffer or the dictionary fills up. Queries that exceed this limit will fail. Set to zero to disable disk spilling.|0 (disabled)|
+
+
+##### Configurations for groupBy v1
+
+|Server runtime property (prefixed by `druid.query.groupBy.`)|Query context parameter|Description|Default|
+|-----------------------|-----------------------|-----------|-------|
+|`maxIntermediateRows`|`maxIntermediateRows`|Maximum number of intermediate rows for the per-segment grouping engine. This is a tuning parameter that does not impose a hard limit; rather, it potentially shifts merging work from the per-segment engine to the overall merging index. Queries that exceed this limit will not fail.|50000|
+|`maxResults`|`maxResults`|Maximum number of results. Queries that exceed this limit will fail.|500000|
+
+
+#### Advanced configurations
+
+##### Common configuragions for all groupBy strategies
+
+|Server runtime property (prefixed by `druid.query.groupBy.`)|Query context parameter|Description|Default|
+|-----------------------|-----------------------|-----------|-------|
+|`defaultStrategy`|`groupByStrategy`|Default groupBy query strategy.|v2|
+|`singleThreaded`|`groupByIsSingleThreaded`|Merge results using a single thread.|false|
+
+
+##### GroupBy v2 configurations
+
+|Server runtime property (prefixed by `druid.query.groupBy.`)|Query context parameter|Description|Default|
+|-----------------------|-----------------------|-----------|-------|
+|`bufferGrouperInitialBuckets`|`bufferGrouperInitialBuckets`|Initial number of buckets in the off-heap hash table used for grouping results. Set to 0 to use a reasonable default.|0|
+|`bufferGrouperMaxLoadFactor`|`bufferGrouperMaxLoadFactor`|Maximum load factor of the off-heap hash table used for grouping results. When the load factor exceeds this size, the table will be grown or spilled to disk. Set to 0 to use a reasonable default.|0|
+|`forceHashAggregation`|`forceHashAggregation`|Force to use hash-based aggregation.|false|
+|`intermediateCombineDegree`|`intermediateCombineDegree`|Number of intermediate nodes combined together in the combining tree. Higher degrees will need a less numer of threads which might be helpful to improve the query performance by reducing the overhead of too many threads if the server has sufficiently powerful cpu cores.|8|
+|`numParallelCombineThreads`|`numParallelCombineThreads`|Hint for the number of parallel combining threads. This should be larger than 1 to turn on the parallel combining feature. The actual number of threads used for parallel combining is min(`druid.query.groupBy.numParallelCombineThreads`, `druid.processing.numThreads`).|1 (disabled)|
```

Comment:
"The actual number of threads used for parallel combining is min(`druid.query.groupBy.numParallelCombineThreads`, `druid.processing.numThreads`)" -- unless the value is 1? IMO too complicated way to set property. Maybe add a separate boolean property, and require numParallelCombineThreads > 1, if configured.

Reply:
Hmm, adding another property to turn on parallel combining looks more complicated to me. 
I added a check that druid.query.groupBy.numParallelCombineThreads cannot be larger than druid.processing.numThreads instead of calculating min of them.